src/compile/utils.ts

```diff
@@ -342,6 +354,7 @@ export function compileSourceString(
     sourceCode: string,
     version: string | CompilerVersionSelectionStrategy,
     remapping: string[],
+    compilationOutput: CompilationOutput[] = [CompilationOutput.ALL],
```

Comment:
If we would move `compilationOutput` after `compilerSettings`, then there will be less compatibility breaks. Other modified functions are internal, so they not cause breaks. If we swap these arguments, we would be able to do minor version bump instead of major.

Reply:
On second thought I would actually like to keep them in this order. My motivation is:

1. `compilationOutput` and `compilerSettings` are introduced almost at the same time (they will differ only be 1 patch version in solc-typed-ast and scribble). So I don't think there will be consumers that know about one, and not the other

2. I think consumers will specify  `compilationOutput` more often than `compilerSettings`. So I would rather keep that one first, so that they don't have to insert 'undefined'.

3. Since `compilationOutput` is of type array, and `compilerSettings` is of type any (but is usualy an object), with the current order, if a user forgets to put in `compilationOutput` but instead directly passes `compilerSettings`, they will get a compiler error. If we switch the order they will not get a compiler error.